public/scripts/sysprompt.js

```diff
@@ -0,0 +1,236 @@
+    }
+
+    const quiet = isTrueBoolean(args?.quiet);
+    const instructNames = system_prompts.map(preset => preset.name);
+    const fuse = new Fuse(instructNames);
+    const result = fuse.search(name);
+
+    if (result.length === 0) {
+        !quiet && toastr.warning(`System prompt "${name}" not found`);
+        return '';
+    }
+
+    const foundName = result[0].item;
+    $select.val(foundName).trigger('input');
+    !quiet && toastr.success(`System prompt "${foundName}" selected`);
+    return foundName;
+}
+
+jQuery(function () {
+    $enabled.on('input', function () {
+        power_user.sysprompt.enabled = !!$(this).prop('checked');
+        toggleSystemPromptDisabledControls();
+        saveSettingsDebounced();
+    });
+
+    $select.on('input', async function () {
+        if (!power_user.sysprompt.enabled) {
+            $enabled.prop('checked', true).trigger('input');
+        }
+
+        const name = String($(this).val());
+        const prompt = system_prompts.find(p => p.name === name);
+        if (prompt) {
+            $content.val(prompt.content);
+            if (!CSS.supports('field-sizing', 'content')) {
+                await resetScrollHeight($content);
+            }
+
+            power_user.sysprompt.name = name;
+            power_user.sysprompt.content = prompt.content;
+        }
+        saveSettingsDebounced();
+    });
+
+    $content.on('input', function () {
+        power_user.sysprompt.content = String($(this).val());
+        saveSettingsDebounced();
+    });
+
+    SlashCommandParser.addCommandObject(SlashCommand.fromProps({
+        name: 'sysprompt',
+        aliases: ['system-prompt'],
+        callback: selectSystemPromptCallback,
+        returns: 'current prompt name',
+        namedArgumentList: [
+            SlashCommandNamedArgument.fromProps({
+                name: 'quiet',
+                description: 'Suppress the toast message on prompt change',
+                typeList: [ARGUMENT_TYPE.BOOLEAN],
+                defaultValue: 'false',
+                enumList: commonEnumProviders.boolean('trueFalse')(),
+            }),
+            SlashCommandNamedArgument.fromProps({
+                name: 'forceGet',
+                description: 'Force getting a name even if system prompt is disabled',
+                typeList: [ARGUMENT_TYPE.BOOLEAN],
+                defaultValue: 'false',
+                enumList: commonEnumProviders.boolean('trueFalse')(),
+            }),
+        ],
+        unnamedArgumentList: [
+            SlashCommandArgument.fromProps({
+                description: 'system prompt name',
+                typeList: [ARGUMENT_TYPE.STRING],
+                enumProvider: () => system_prompts.map(x => new SlashCommandEnumValue(x.name, null, enumTypes.enum, enumIcons.preset)),
+            }),
+        ],
+        helpString: `
+            <div>
+                Selects a system prompt by name. Enables the use of system prompt if not already enabled.
```

Comment:
Not true currently. If disabled, this will always return empty. Either fix the wording, or change the if in the callback to allow disabled state when a name is provided.